src/rez/utils/resources.py

```diff
@@ -41,6 +43,15 @@
 from rez.exceptions import ResourceError
 from rez.utils.logging_ import print_debug
 
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    # this is not available in typing until 3.11, but due to __future__.annotations
+    # we can use it without really importing it
+    from typing import Self
```

Comment:
This makes it impossible to run mypy with python < 3.11 right?

Reply:
as I explained in my other comment code within an `if TYPE_CHECKING` does not actually run.

Reply:
Note how I said "run mypy". Like, if we were to run mypy with `python_version = 3.7` for example.